pyogrio/_ogr.pyx

```diff
@@ -323,3 +323,34 @@ def _get_driver_metadata_item(driver, metadata_item):
             metadata = None
 
     return metadata
+
+
+def _get_drivers_for_path(path):
+    cdef OGRSFDriverH driver = NULL
+    cdef int i
+    cdef char *name_c
+
```

Comment:
Maybe nice to also add support for ".shp.zip" and ".gpkg.zip" as it is also supported in [gdal commonutils](https://github.com/OSGeo/gdal/blob/949ac934a6518f0e4fe60ca55827ec5e9d29d755/apps/commonutils.cpp#L69)?